ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/console/params/ParametersService.java

```diff
@@ -0,0 +1,61 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.console.params;
+
+import org.apache.commons.lang.StringUtils;
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class ParametersService {
+
+    private ParametersProvider parametersProvider;
+    public static final ObjectMapper MAPPER = configureMapper();
+    public static final JsonFactory FACTORY = MAPPER.getJsonFactory();
+
+    public void registerParametersProvider(final ParametersProvider parametersProvider) {
+        if (this.parametersProvider != null) {
+            throw new IllegalStateException("Parameters provider already registered");
+        }
+        this.parametersProvider = parametersProvider;
+    }
+
+    public Map<String, Object> getParameters() throws Exception {
+        if (parametersProvider == null || StringUtils.isBlank(parametersProvider.getParametersJson())) {
+            return Collections.emptyMap();
+        }
+
+        JsonParser parser = FACTORY.createJsonParser(parametersProvider.getParametersJson());
+        JsonNode node = MAPPER.readTree(parser);
+        return mapFirstLevelChildren(node);
```

Comment:
You can convert tree to map directly using jackson.

Code from google:
```
parser.readValueAs(new TypeReference<Map<String, Object>[]>() {});
```

Or other example:
```
Map<String, Object> carMap = mapper.readValue(new File(
                    "result.json"), new TypeReference<Map<String, Object>>() {
            });
```
SO: http://stackoverflow.com/questions/24501372/convert-json-to-hash-map-with-jackson-java